src/ostree/ot-builtin-summary.c

```diff
@@ -217,7 +217,10 @@ ostree_builtin_summary (int argc, char **argv, GCancellable *cancellable, GError
       if (opt_raw)
         flags |= OSTREE_DUMP_RAW;
 
-      summary_data = ot_file_mapat_bytes (repo->repo_dir_fd, "summary", error);
+      glnx_fd_close int fd = -1;
+      if (!glnx_openat_rdonly (repo->repo_dir_fd, "summary", TRUE, &fd, error))
+        return FALSE;
+      summary_data = ot_fd_readall_or_mmap (fd, 0, error);
```

Comment:
This seems like a common enough pattern to offer an equivalent wrapper in `ot-fs-util.c`, no?

Reply:
Yeah, there are a few places, but it's messy since the summary path wants `GBytes*`, the detached metadata wants `GVariant` right now, and I didn't quite feel like two more wrappers.

Reply:
Er, sorry, I thought you meant something else.  Eh...there's just two instances of that right now right?

Reply:
Oh right sorry, I meant the other one (`openat` followed by `ot_variant_read_fd`). We can do that later on though.